application/helpers/expressions/em_manager_helper.php

```diff
@@ -9972,7 +9935,34 @@ static public function ShowSurveyLogicFile($sid, $gid=NULL, $qid=NULL,$LEMdebugL
                 // FINALLY, SHOW THE QUESTION ROW(S), COLOR-CODING QUESTIONS THAT CONTAIN ERRORS
                 //////
                 $errclass = ($errorCount > 0) ? 'danger': '';
-                $errText=($errorCount > 0) ? "<br><em class='label label-danger'>".$LEM->ngT("This question has at least {n} error.|This question has at least {n} errors.",$errorCount)."<em>" : "";
+                $errText=($errorCount > 0) ? "<br><em class='label label-danger'>".$LEM->ngT("This question has at least {n} error.|This question has at least {n} errors.",$errorCount)."</em>" : "";
+                /* Construct the warnings */
+                $sWarningsText = "";
+                if(count($aWarnings) > 0) {
+                    $sWarningsText .= "<div class='alert alert-warning'>";
+                    $sWarningsText .= "<strong class=''>".$LEM->ngT("This question has at least {n} warning.|This question has at least {n} warnings.",count($aWarnings))."</strong>";
+                    $sWarningsText .= "<ul class='list-unstyled small text-warning'>";
+                    $warningsDone = array();
+                    foreach($aWarnings as $aWarning) {
+                        if(!in_array($aWarning[0],$warningsDone) ) {
+                            $sWarningsText .= "<li>";
+                            if(!empty($aWarning[2])) {
```

Comment:
Here's a classic if-statement that could be replaced with inheritance.

Reply:
. Unsure : warnings can have a link or can not have a link :) It's still need to check if link is OK or not.

Here : we can use `null !== $aWarning[2]` because it set as null by default.